src/catalog/catalog_entry/table_catalog_entry.cpp

```diff
@@ -186,3 +201,17 @@ unique_ptr<CreateTableInfo> TableCatalogEntry::Deserialize(Deserializer &source)
 	}
 	return info;
 }
+
+unique_ptr<CatalogEntry> TableCatalogEntry::Copy(ClientContext &context) {
+	auto create_info = make_unique<CreateTableInfo>(schema->name, name);
```

Comment:
What about table constraints?

Reply:
Copying constrains is trick because they are unique_ptr. Should I change to shared_ptr?

Reply:
Constraints have a Copy method (the regular Constraint, not the BoundConstraint). You can use that to copy them.